srml/support/src/dispatch.rs

```diff
@@ -67,38 +67,114 @@ pub trait Parameter: Codec + Clone + Eq {}
+/// # use srml_support::dispatch::Result;
+/// # use srml_system::{self as system, Trait, ensure_signed};
+/// 
+/// decl_module! {
+/// 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+/// 		fn my_function(origin) {
+///				// Your implementation
+/// 		}
+///		}
+/// }
+/// # fn main() {}
+/// ```
+///
+/// The macro automatically expands a shorthand function declaration to return the `Result` type.
 ///
-/// where "Trait" is a trait describing this module's requirements for the Runtime type.
-/// T::Origin is declared using a impl_outer_origin! per-module macro (which is generated by the
-/// construct_runtime! macro) and automatically includes all the modules that are used by the
-/// runtime (alongside with a variant called "system").
+/// ## Priviledged Function Example
 ///
-/// A runtime module is a collection of functions unified by a common problem domain and certain
-/// shared types. The "functions" do not actually return values (see Dispatchable) and are only
-/// used for side effects.
+/// ```rust
+/// # #[macro_use]
+/// # extern crate srml_support;
+/// # use srml_support::dispatch::Result;
+/// # use srml_system::{self as system, Trait, ensure_signed};
 ///
-/// For every module an associated enum (usually "Call") is generated with every variant
-/// corresponding to a function of the module. This enum implements Callable and thus its values
-/// can be used as an extrinsic's payload.
+/// decl_module! {
+/// 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+///			fn my_function() -> Result {
+///				// Your implementation
+/// 			Ok(())
+/// 		}
+///		}
+/// }
+/// # fn main() {}
+/// ```
 ///
-/// The `on_initialize` and `on_finalize` functions are special, since it can either take no
-/// parameters, or one parameter, which has the runtime's block number type.
+/// If the `origin` param is omitted, the macro adds it as the first parameter and adds `ensure_root(origin)` as the first line of the function.
 ///
-/// ### Module with instances
+/// ## Multiple Module Instances Example
 ///
-/// decl_module! support modules with instances with the following syntax: (DefaultInstance type is
-/// optionnal)
-/// ```nocompile
-/// pub struct Module<T: Trait<I>, I: Instance = DefaultInstance> for enum Call where origin: T::Origin {}
+/// ```rust
+/// # #[macro_use]
+/// # extern crate srml_support;
+/// # use srml_support::dispatch::Result;
+/// # use srml_system::{self as system, ensure_signed};
+/// # pub struct DefaultInstance;
+/// # pub trait Instance {}
+/// # impl Instance for DefaultInstance {}
+/// 
+/// pub trait Trait<I: Instance=DefaultInstance>: system::Trait {}
+/// 
+/// decl_module! {
+/// 	pub struct Module<T: Trait<I>, I: Instance = DefaultInstance> for enum Call where origin: T::Origin {
+/// 		// Your implementation
+/// 	}
+/// }
+/// # fn main() {}
 /// ```
+///
+/// This macro supports modules with multiple instances. See [balances](../srml_balances/index.html) example.
```

Comment:
If there is a better description + your example, then this line can be removed.

Reply:
@nczhu I gave this a shot. We may also need an example, or a link to an example of how you would actually declare multiple instantiations in the `construct_runtime!` docs.

Reply:
Sounds good your description :)